test/toy_class_test.rb

```diff
@@ -17,13 +17,6 @@
 
     describe "getting a class's superclass" do
       specify 'returns the superclass' do
-        class_a = klass.public_send(@new_method)
-        @class = klass.public_send(@new_method, class_a)
-
-        assert_equal class_a, call_method(:superclass)
-      end
```

Comment:
I disagree that this test was extraneous. It does look superficially similar to the genuinely extraneous one in `toy_object_test.rb`, but there’s a difference.

While the API provides only one way to create an _object_ (i.e. calling `.new` on a class), it gives us two ways to create a _class_:

* calling `Class.new()`, where the resulting class should get `Class` (the default) as its superclass; or
* calling `Class.new(my_class)`, where the resulting class should get `my_class` as its superclass.

So we need these two separate tests to cover these two uses of the `Class.new` API, whereas we only need one test to check the “`MyClass.new.class == MyClass`” property of the `Object#class` API.

Reply:
Ah, this is my bad! I'll bring this back 😄